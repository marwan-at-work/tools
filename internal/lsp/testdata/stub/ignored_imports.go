package stub

import (
	"context"
	. "time"
	_ "time"
)

// This file tests that dot-imports and underscore imports
// are properly ignored and that a new import is added to
// refernece method types

var (
	_                 = Time{}
	_ context.Context = (*ignoredContext)(nil) //@suggestedfix("(", "quickfix")
)

type ignoredContext struct{}
