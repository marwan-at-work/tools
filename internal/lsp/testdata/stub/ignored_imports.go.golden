-- suggestedfix_ignored_imports_15_22 --
package stub

import (
	"context"
	"time"
	. "time"
	_ "time"
)

// This file tests that dot-imports and underscore imports
// are properly ignored and that a new import is added to
// refernece method types

var (
	_                 = Time{}
	_ context.Context = (*ignoredContext)(nil) //@suggestedfix("(", "quickfix")
)

type ignoredContext struct{}

// Deadline implements context.Context
func (*ignoredContext) Deadline() (deadline time.Time, ok bool) {
	panic("unimplemented")
}

// Done implements context.Context
func (*ignoredContext) Done() <-chan struct{} {
	panic("unimplemented")
}

// Err implements context.Context
func (*ignoredContext) Err() error {
	panic("unimplemented")
}

// Value implements context.Context
func (*ignoredContext) Value(key interface{}) interface{} {
	panic("unimplemented")
}

